src/inject.ts

```diff
@@ -0,0 +1,620 @@
+		});
+	}
+	summary.ALREADY_EXISTS++;
+	summary.INCOMPLETE_CANDIDATES += isComplete ? 0 : 1;
+	return isComplete;
+}
+
+function injectionSuccess(
+	progress: string,
+	filePathLog: string,
+	injectionResult: InjectionResult,
+	summary: InjectSummary,
+	matchedSearchee: SearcheeWithLabel,
+	matchedDecision: DecisionAnyMatch,
+	meta: Metafile,
+	tracker: string,
+): void {
+	logger.info({
+		label: Label.INJECT,
+		message: `${progress} Injected ${filePathLog} - ${chalk.green(injectionResult)}`,
+	});
+	sendResultsNotification(matchedSearchee!, [
+		[
+			{ decision: matchedDecision!, metafile: meta },
+			tracker,
+			injectionResult,
+		],
+	]);
+	summary.INJECTED++;
+	if (matchedDecision! === Decision.MATCH_PARTIAL) {
+		summary.PARTIAL_MATCHES++;
+	} else {
+		summary.FULL_MATCHES++;
+	}
+}
+
+async function injectTorrentFiles(
+	torrentFilePaths: string[],
+	summary: InjectSummary,
+	options: { cleanUpOldTorrents: boolean },
+): Promise<void> {
+	const searchees = await findAllSearchees(Label.INJECT);
+
+	let count = 0;
+	for (const torrentFilePath of torrentFilePaths) {
+		const progress = chalk.blue(`(${++count}/${torrentFilePaths.length})`);
+		const filePathLog = getTorrentFilePathLog(torrentFilePath);
+		let meta: Metafile;
+		try {
+			meta = await parseTorrentFromFilename(torrentFilePath);
+		} catch (e) {
+			logger.error({
+				label: Label.INJECT,
+				message: `${progress} Failed to parse ${filePathLog}`,
+			});
+			logger.debug(e);
+			deleteTorrentFileIfElligible(torrentFilePath, {
+				isComplete: false,
+				cleanUpOldTorrents: options.cleanUpOldTorrents,
+			});
+			continue;
+		}
+		const metaLog = getLogString(meta, chalk.bold.white);
+
+		const torrentNameInfo =
+			await parseInfoFromSavedTorrent(torrentFilePath);
+		const tracker = torrentNameInfo?.tracker ?? UNKNOWN_TRACKER;
+		if (tracker === UNKNOWN_TRACKER) {
+			summary.FOUND_BAD_FORMAT = true;
+		}
+
+		const { fullMatches, partialMatches, foundBlocked } =
+			await injectDecideStage(meta, searchees);
+		const matches: AllMatches = [...fullMatches, ...partialMatches];
+		if (!matches.length) {
+			if (foundBlocked) {
+				logger.info({
+					label: Label.INJECT,
+					message: `${progress} ${metaLog} ${chalk.yellow("possibly blocklisted")}: ${filePathLog}`,
+				});
+				summary.BLOCKED++;
+			} else {
+				logger.info({
+					label: Label.INJECT,
+					message: `${progress} ${metaLog} ${chalk.red("has no matches")}: ${filePathLog}`,
+				});
+				summary.UNMATCHED++;
+			}
+			deleteTorrentFileIfElligible(torrentFilePath, {
+				isComplete: false,
+				cleanUpOldTorrents: options.cleanUpOldTorrents,
+			});
+			continue;
+		}
+
+		const {
+			injectionResult,
+			matchedSearchee,
+			matchedDecision,
+			linkedNewFiles,
+		} = await injectInitialAction(meta, matches, tracker);
+		if (injectionResult === InjectionResult.FAILURE) {
+			injectionFailed(progress, filePathLog, injectionResult, summary);
+			continue; // Never delete failed torrent files
```

Comment:
why never delete failed torrent files?

Reply:
If I recall, this failure is potentially due to incomplete torrents and there is a cleanup done independently after a week of age.

Don't quote me, this is just what I think I remember.

Reply:
I added this a few days ago as injection failures is always transient or user error.